src/server/api/search.ts

```diff
@@ -0,0 +1,38 @@
+import Express from 'express'
+import { createValidator } from 'express-joi-validation'
+import Joi from '@hapi/joi'
+import rateLimit from 'express-rate-limit'
+import { container } from '../util/inversify'
+import { DependencyIds } from '../constants'
+import { SearchControllerInterface } from '../controllers/interfaces/SearchControllerInterface'
+import { SearchResultsSortOrder } from '../repositories/enums'
+
+const urlSearchRequestSchema = Joi.object({
+  query: Joi.string().required(),
+  order: Joi.string()
+    .required()
+    .allow(...Object.values(SearchResultsSortOrder))
+    .only(),
+  limit: Joi.number(),
+  offset: Joi.number(),
+})
+
+const apiLimiter = rateLimit({
+  windowMs: 1000, // 1 second
+  max: 20,
+})
```

Comment:
need to set `keyGenerator` option with our `getIp` function, since we're proxied behind Cloudflare on staging/production which means `req.ip` will be the Cloudflare IP by default

Reply:
can we also write to the logs when this is triggered, via the `onLimitReached` option? we should log the IP address in addition to the other metrics provided